### PR TITLE
Calico with Mesos and Kubernetes integrated

### PR DESCRIPTION
**Pull request by Zogg**
After a long period, with issues resolved, we finally have the Calico and Mesos and Kubernetes working together.

*Champagne popping sound*

Calico: 0.15.0
Kubernetes: 1.2.0
Mesos: 0.27.0
Marathon: 1.0.0_RC1

Components version upgrades are to be supplied through their respective PRs.

### CODE REVIEW

**Comment by siddharthist** [comment]
Fixes #1514

File: test.mesos.yml [hunk 1]
```diff
@@ -0,0 +1,138 @@
+---
```
Comment (siddharthist) [inline test.mesos.yml]: I think this was probably checked in accidentally

File: terraform/gce/gce.tf [hunk 2]
```diff
@@ -57,6 +57,10 @@ resource "google_compute_firewall" "mi-firewall-internal" {
   source_ranges = ["${google_compute_network.mi-network.ipv4_range}"]
 
   allow {
+    protocol = "4"
```
Comment (siddharthist) [inline terraform/gce/gce.tf]: What is the "4" protocol?

**Comment by siddharthist** [comment]
@Zogg Awesome job! Do we need to update/create any documentation around this?

File: terraform/gce/gce.tf (hunk 2 above)
Comment (Zogg) [inline terraform/gce/gce.tf]: Allows `ipip` packets through the network, if I recall correctly.

File: terraform/gce/gce.tf [hunk 3]
```diff
@@ -198,8 +198,10 @@ resource "google_compute_instance" "mi-worker-nodes" {
   }
 }
 
+
```
Comment (siddharthist) [inline terraform/gce/gce.tf]: We can probably just drop this commit

**Comment by siddharthist** [comment]
@Zogg Also, have you been able to test with more recent versions of Mesos and Marathon? Marathon is at v1.1.1, and they recommend Mesos 0.28.0 to go with it.

**Comment by Zogg** [comment]
A lot of testing has been done with various mesos/marathon/calico/kube versions. The exact versions that will go into 1.2 will be resolved in respective component PRs.

**Pull request merged**